### PR TITLE
Feature/smart search

### PR DESCRIPTION
**Pull request by Aydoun**
* Include fuse.js for smart searching
* Types needs correction from reviewers

### CODE REVIEW

File: src/components/Search/index.tsx [hunk 1]
```diff
@@ -1,96 +1,39 @@
...
+import Groupby from 'lodash.groupby';
+
+import {
+  Container,
+  SearchIcon,
+  Input,
+  DropDownContainer,
+  InputContainer,
+  RingName,
+  Technology,
```
Comment (perlo27) [inline src/components/Search/index.tsx]: We have same type name, so when you import this component and try to use the type same type name on line 32 it uses the component type. 
so just rename component name or use  ```Technology as TechnologyComponent```

File: src/components/Search/index.tsx [hunk 2]
```diff
@@ -99,28 +42,18 @@ export const Search: React.FC<SearchProps> = ({ setHighlighted }) => {
...
-              },
-            ),
-    [value, technologies],
-  );
+  const data = useMemo(() => {
+    if (value.length < 2) return [];
+
+    const fuse = new Fuse(technologies, options);
+    // @ts-ignore
+    const results: fuseResult[] = fuse.search(value);
```
Comment (perlo27) [inline src/components/Search/index.tsx]: no need to specify type here, it infers type perfectly

File: package.json [hunk 3]
```diff
@@ -4,6 +4,8 @@
   "private": true,
   "dependencies": {
     "d3": "^5.15.0",
+    "fuse.js": "^5.0.7-beta",
```
Comment (perlo27) [inline package.json]: So it wasn't your mistake with typings, it was an issue with typings in their repo,
they fixed it today though, what a luck :)
just change version to ```^5.0.9-beta```

File: src/components/Search/index.tsx [hunk 4]
```diff
@@ -99,28 +42,18 @@ export const Search: React.FC<SearchProps> = ({ setHighlighted }) => {
...
+    return Groupby(
```
Comment (perlo27) [inline src/components/Search/index.tsx]: nice improvement 👍

File: src/components/Search/index.tsx (hunk 1 above)
Comment (perlo27) [reply src/components/Search/index.tsx]: Or better we should follow general convention regarding naming of types, in this case type for Technology should be ```ITechnology```

File: package.json (hunk 3 above)
Comment (Aydoun) [reply package.json]: that solved eveything haha

**Pull request merged**